### PR TITLE
Add designate config override

### PR DESCRIPTION
**Pull request by ktibi**

### CODE REVIEW

File: ansible/roles/kolla-openstack/vars/main.yml [hunk 1]
```diff
@@ -19,6 +19,11 @@ kolla_openstack_custom_config:
     dest: "{{ kolla_node_custom_config_path }}/cinder"
     patterns: "*"
     enabled: "{{ kolla_enable_cinder }}"
+  # Designate.
+  - src: "{{ kolla_extra_config_path }}/designate"
+    dest: "{{ kolla_node_custom_config_path }}/designate"
+    patterns: "*"
+    enabled: "{{ kolla_enable_designate }}"
```
Comment (markgoddard) [inline ansible/roles/kolla-openstack/vars/main.yml:26]: This is fine for `{{ kolla_extra_config_path }}/designate` but will not template `{{ kolla_extra_config_path }}/designate.conf`. Is that ok?

Comment (ktibi) [reply ansible/roles/kolla-openstack/vars/main.yml:26]: for me it's ok but I'll add for designate.conf too

Comment (markgoddard) [reply ansible/roles/kolla-openstack/vars/main.yml:26]: Also need to add ansible/roles/kolla-openstack/templates/designate.conf.j2. Please test these changes.

**Comment by ktibi** [comment]
@markgoddard Can you merge ?

File: ansible/kolla-openstack.yml [hunk 2]
```diff
@@ -100,6 +100,7 @@
           with_items:
             - { name: ceph, file: ceph.conf }
             - { name: cinder, file: cinder.conf }
+            - { name: designate, file: designate.conf }
```
Comment (markgoddard) [inline ansible/kolla-openstack.yml:103]: Need to add this at the bottom of this file:

`kolla_extra_designate: "{{ kolla_extra_config.designate | default }}"`

**Review by markgoddard — CHANGES_REQUESTED** [review]
Almost there.

File: ansible/kolla-openstack.yml (hunk 2 above)
Comment (ktibi) [reply ansible/kolla-openstack.yml:103]: it's done.

Comment (markgoddard) [reply ansible/kolla-openstack.yml:103]: Thanks.

**Review by markgoddard — APPROVED** [review]

**Pull request merged**